### ISSUE TITLE
OSS Nomad spam Enterprise-only related errors in logs

### ISSUE THREAD

**Issue by Oloremo** [hashicorp/nomad#9827]
### Nomad version
`Nomad v1.0.2 (4c1d4fc6a5823ebc8c3e748daec7b4fda3f11037)`

### Operating system and Environment details
Centos 7.6 x64

### Issue
Nomad logs are spammed by:
```
2021-01-15T14:52:09.436Z [ERROR] http: request failed: method=GET path=/v1/operator/license error="Nomad Enterprise only endpoint" code=501
```

We use OSS version so not sure why it does it.

Maybe related to the fact that we have telemetry enabled?

### Reproduction steps

1. Run OSS version
2. Wait

Kinda related to https://github.com/hashicorp/nomad/issues/4567

**Comment by drewbailey** [comment]
Hi @Oloremo thanks for reporting, This seems to be caused by the UI checking for enterprise functionality. Unfortunately it looks like it checks or tried to fetch license info on every request. We'll take a look and discuss how to improve this.

**Comment by backspace** [comment]
hmm, I believe the UI only [attempts to load the license once](https://github.com/hashicorp/nomad/blob/master/ui/app/routes/application.js#L40), when the application initialises.

Unfortunately, as the API is currently set up, there’s no way for the UI to know whether it’s communicating with enterprise vs OSS without this query. One possible avenue of improvement would be to add an endpoint that doesn’t return an error code when it’s OSS, or adjusting this one to have a defined non-error response that represents OSS. `204 No Content`?

**Comment by drewbailey** [comment]
Sorry @backspace, "on every request" I meant I was seeing it each time I loaded http://localhost:4646/ui/jobs. I like your suggested approach, the API should be able to provide a better response.

**Comment by backspace** [comment]
If you haven’t already done anything to this effect, @drewbailey, I could dabble in a little Go to hopefully make this happen 😯

**Comment by drewbailey** [comment]
@backspace I haven't had the chance to dig in yet, happy to help out if you decide to dabble :grin:

**Comment by backspace** [comment]
Will do, I have a few things I should wrap up before starting something knew, but I’ll let you know 😀
